api/src/test/java/org/openmrs/validator/ObsValidatorTest.java

```diff
@@ -42,6 +42,22 @@
 	@Autowired
 	private ObsValidator obsValidator;
 	
+	/**
+	 * @see ObsValidator#validate(java.lang.Object, org.springframework.validation.Errors)
+	 */
+	@Test
+	public void validate_shouldFailIfPersonAndEncounterObjectsDoNotMatch() {
+		
+		executeDataSet("org/openmrs/validator/include/ObsValidatorTest.xml");
+		Obs obs = Context.getObsService().getObs(7);
+        
```

Comment:
could you show that the  the obs and its encounter point to different persons  for this specific obs your testing ??

Reply:
> could you show that the the obs and its encounter point to different persons for this specific obs your testing ??

when you look at the data set i created and the test i made,it covers that

Reply:
but you need to show it here ,
we almost never check to see the test Dataset

Reply:
@HerbertYiga , add more clarification about the obs here and its encounter  .
are they having diff patients ?? 
because the `obs.person_id = 7 = encounter.patient_id.` in your dataset

Reply:
@HerbertYiga , can you assert that the obs object here and its coresponding encounter point to different persons??